src/messenger.rs

```diff
@@ -205,6 +201,12 @@ pub enum SaslError {
     #[error("Sasl session error: {0}")]
     SaslSessionError(#[from] SessionError),
 
+    #[error("Invalid SASL config: {0}")]
+    InvalidConfig(#[from] SASLError),
+
+    #[error("Other SASL error: {0}")]
+    Other(String),
```

Comment:
I guess you've introduced this variant so that you have a catch-all version for the callback? I'm wondering if the callback should even use `SaslError` in the first place or if it should rather return `Box<dyn std::error::Error + Send + Sync>` and you convert this to a `SaslError` variant like

```rust
#[derive(Error, Debug)]
pub enum SaslError {
    // ...

    #[error("OAuth Callback: {0}")]
    OauthCallback(Box<dyn std::error::Error + Send + Sync>),

    // ...
}
```

Reply:
That's probably a good idea. I'll change it :+1: